### PR TITLE
Use udisks2 to prepare external data drive

### PR DESCRIPTION
**Pull request by agners**
Make use of udisks2 to find, partition and format the external data
drive. This should lead to better error handling and ease additional
features in the future.

### CODE REVIEW

**Review by pvizeli — APPROVED** [review]
Nice!

**Comment by pvizeli** [comment]
Remember me to add some linter action :)

File: datadisk.go [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 import (
-	"bytes"
+	"context"
 	"fmt"
 	"log"
 	"os"
-	"os/exec"
+	"strings"
+
+	"github.com/home-assistant/os-agent/udisks2"
```
Comment (pvizeli) [inline datadisk.go:10]: is that missing in the commit?

Comment (agners) [reply datadisk.go:10]: Indeed! Fixed!

**Pull request merged**